notation.go

```diff
@@ -52,10 +55,23 @@ type Signer interface {
 // remote.
 // The descriptor of the sign content is returned upon sucessful signing.
 func Sign(ctx context.Context, signer Signer, repo registry.Repository, opts SignOptions) (ocispec.Descriptor, error) {
-	targetDesc, err := repo.Resolve(ctx, opts.ArtifactReference)
+	logger := log.GetLogger(ctx)
+
+	artifactRef := opts.ArtifactReference
+	targetDesc, err := repo.Resolve(ctx, artifactRef)
 	if err != nil {
 		return ocispec.Descriptor{}, err
 	}
+	if !isDigestReference(artifactRef) {
+		// artifactRef is not a digest reference
+		ref, err := orasRegistry.ParseReference(artifactRef)
+		if err != nil {
+			return ocispec.Descriptor{}, err
+		}
+		logger.Warnf("Always sign the artifact using digest(`@sha256:...`) rather than a tag(`:%s`) because tags are mutable and a tag reference can point to a different artifact than the one signed", ref.ReferenceOrDefault())
```

Comment:
`oras-go` will not use the default `latest`. Therefore, it should be `ref.Reference` instead of `ref.ReferenceOrDefault()`.